board/safety/safety_subaru.h

```diff
@@ -40,19 +45,34 @@ static void subaru_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
       controls_allowed = 0;
     }
     subaru_cruise_engaged_last = cruise_engaged;
+  } else if ((addr == 0x144) && (bus_number == 0)) {
+    int cruise_engaged = (to_push->RDHR >> 17) & 1;
+    if (cruise_engaged && !subaru_cruise_engaged_last) {
+      controls_allowed = 1;
+    } else if (!cruise_engaged) {
+      controls_allowed = 0;
+    }
+    subaru_cruise_engaged_last = cruise_engaged;
   }
 }
 
 static int subaru_tx_hook(CAN_FIFOMailBox_TypeDef *to_send) {
   uint32_t addr = to_send->RIR >> 21;
 
   // steer cmd checks
-  if (addr == 0x122) {
-    int desired_torque = ((to_send->RDLR >> 16) & 0x1FFF);
+  if (addr == 0x122 || addr == 0x164) {
+    int desired_torque = 0;
+    if (addr == 0x122) {
+    desired_torque = ((to_send->RDLR >> 16) & 0x1FFF);
+    desired_torque = to_signed(desired_torque, 13);
+    } else if (addr == 0x164) {
+    desired_torque = ((to_send->RDLR >> 8) & 0x1FFF);
+    desired_torque = to_signed(desired_torque, 13);
+    }
+    
     int violation = 0;
     uint32_t ts = TIM2->CNT;
-    desired_torque = to_signed(desired_torque, 13);
```

Comment:
This can stay here. Same code in both cases

Reply:
I'll fix it now